### PR TITLE
DOC: Clean up the pie docstring PR

### PR DESCRIPTION
**Pull request by phobson**
Supersedes #8357 

Builds upon @kahnchana 's work with some minor reorganization

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -2469,105 +2469,105 @@ def stem(self, *args, **kwargs):
...
-        Return value:
-          If *autopct* is *None*, return the tuple (*patches*, *texts*):
+        Examples
+        --------
+        The pie chart will probably look best if the figure and axes are
+        square, or the Axes aspect is equal.
 
-            - *patches* is a sequence of
-              :class:`matplotlib.patches.Wedge` instances
+        .. plot:: mpl_examples/pie_and_polar_charts/pie_demo_features.py
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: Soon, we won't need this anymore :D

File: lib/matplotlib/tests/test_backend_pgf.py [hunk 2]
```diff
@@ -81,9 +81,13 @@ def create_figure():
 
 
 # test compiling a figure to pdf with xelatex
+<<<<<<< HEAD
 @needs_xelatex
 @pytest.mark.style('default')
 @pytest.mark.backend('pgf')
+=======
```
Comment (NelleV) [inline lib/matplotlib/tests/test_backend_pgf.py]: There's a problem in this file.

**Comment by phobson** [comment]
@NelleV I forgot that the test_backend_pgf file had been modified in the original PR. I took that out along with the other extraneous whitespace changes

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: @NelleV curious: what does the future look like in this regard?

**Comment by NelleV** [comment]
Feel free to self-merge once the tests pass. This has been approved already in the previous Pr.

File: lib/matplotlib/axes/_axes.py (hunk 1 above)
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: On matplotlib, I am not sure, as our pyplot API page is gigantic, but on sklearn it looks like this:
http://scikit-learn.org/stable/modules/generated/sklearn.isotonic.IsotonicRegression.html#sklearn.isotonic.IsotonicRegression

If you scroll down, you'll see all examples using the isotonic regression (there's only one).
I am not sure how to activate this feature just yet, but that is my next step :)

Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: oooooooh fancy!

**Comment by phobson** [comment]
@NelleV Does this error from the doc build mean anything to you?

```
/home/travis/build/matplotlib/matplotlib/lib/matplotlib/axes/_axes.py:docstring of matplotlib.axes.Axes.pie:97: ERROR: Error in "rubric" directive:
no content permitted.

.. rubric:: Examples

        .. plot:: mpl_examples/pie_and_polar_charts/pie_demo_features.py

Building HTML failed.
```

**Comment by NelleV** [comment]
It's an indentation problem. That's all I can say :(

**Comment by phobson** [comment]
Ok, I'll dig into that later

**Comment by QuLogic** [comment]
The previous PR was never approved, so don't self-merge.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -2682,9 +2684,9 @@ def get_next_color():
             self.set_frame_on(False)
 
             self.set_xlim((-1.25 + center[0],
-                            1.25 + center[0]))
+                          1.25 + center[0]))
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: This is not indented correctly.

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -2469,105 +2469,107 @@ def stem(self, *args, **kwargs):
...
+        counterclockwise, by default starting from the x-axis.
 
-        Keyword arguments:
+        Parameters
+        ----------
+        x : array-like
+            The input array used to make the pie chart.
 
-          *explode*: [ *None* | len(x) sequence ]
+        explode : array-like, optional (default is None)
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: I think our convention has been `optional, default: None`.

File: lib/matplotlib/axes/_axes.py [hunk 5]
```diff
@@ -2469,105 +2469,107 @@ def stem(self, *args, **kwargs):
...
             If not *None*, is a string or function used to label the wedges
             with their numeric value.  The label will be placed inside the
             wedge.  If it is a format string, the label will be ``fmt%pct``.
             If it is a function, it will be called.
 
-          *pctdistance*: scalar
+        pctdistance : float, optional (default is 0.6)
             The ratio between the center of each pie slice and the
             start of the text generated by *autopct*.  Ignored if
             *autopct* is *None*; default is 0.6.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Can delete default here.

File: lib/matplotlib/axes/_axes.py [hunk 6]
```diff
@@ -2469,105 +2469,107 @@ def stem(self, *args, **kwargs):
...
 
-        or::
+        texts : list
+            A is a list of the label :class:`matplotlib.text.Text` instances.
 
-          axes(aspect=1)
+        autotexts : list
+            A is a list of :class:`~matplotlib.text.Text` instances for the
+            numeric labels. Is returned only if parameter *autopct* is
+            not *None*.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Trailing whitespace.

**Comment by phobson** [comment]
@QuLogic  don't worry. wasn't going to after seen that weird "rubric" failure.

File: lib/matplotlib/axes/_axes.py [hunk 7]
```diff
@@ -2469,105 +2469,107 @@ def stem(self, *args, **kwargs):
...
-                         label_namer=None)
+                      label_namer=None)
     def pie(self, x, explode=None, labels=None, colors=None,
             autopct=None, pctdistance=0.6, shadow=False, labeldistance=1.1,
             startangle=None, radius=None, counterclock=True,
             wedgeprops=None, textprops=None, center=(0, 0),
             frame=False, rotatelabels=False):
-        r"""
-        Plot a pie chart.
+        r"""Plot a pie chart.
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Can you try reverting this? I think it might fix it. At least it seems to be working locally...

**Comment by NelleV** [comment]
The indentation problem probably comes from our magic interpolation docstring functions.

File: lib/matplotlib/axes/_axes.py (hunk 7 above)
Comment (phobson) [inline lib/matplotlib/axes/_axes.py]: Question: does it even need to be a raw string?

Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: I don't think so, but if everything breaks when you remove the raw string, then it needs to be a raw string :p

Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: I don't see any backslashes, so maybe not?

**Comment by QuLogic** [comment]
Congrats on attaining AppVeyor build 1000.

**Comment by dstansby** [comment]
Backported to `2.0.x` via https://github.com/matplotlib/matplotlib/commit/5e3df6691797be4a8f57ac31d43fd501d36131ae

**Comment by phobson** [comment]
@kahnchana thanks for this!

**Pull request merged**